sdk/src/main/java/com/karhoo/sdk/analytics/Payloader.kt

```diff
@@ -144,19 +176,140 @@ class Payloader internal constructor(builder: Builder) {
             payload[PAYMENT_DATE] = date.toString()
             return this
         }
 
+        fun checkoutOpened(quoteId: String?): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+
+            return this
+        }
+
+        fun cardAuthorisationSuccess(quoteId: String?): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+
+            return this
+        }
+
+        fun loyaltyStatusRequested(
+            slug: String?,
+            errorMessage: String?,
+            quoteId: String?,
+            correlationId: String?,
+            loyaltyStatus: LoyaltyStatus?,
+            balance: Int?,
+            loyaltyProgramme: LoyaltyProgramme?
+        ): Builder {
+            payload[LOYALTY_ENABLED] =
+                loyaltyStatus?.canBurn == true || loyaltyStatus?.canEarn == true
+
+            payload[LOYALTY_STATUS_SUCCESS] = loyaltyStatus != null
+            loyaltyStatus?.let {
+                it.canBurn?.let { canBurn ->
+                    payload[LOYALTY_STATUS_CAN_BURN] = canBurn
+                }
+                it.canEarn?.let { canEarn ->
+                    payload[LOYALTY_STATUS_CAN_EARN] = canEarn
+                }
+                balance?.let { balance ->
+                    payload[LOYALTY_STATUS_BALANCE] = balance
+                }
+            }
+
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+            loyaltyProgramme?.name?.let {
+                payload[LOYALTY_NAME] = it
+            }
+            correlationId?.let {
+                payload[CORRELATION_ID] = it
+            }
+            slug?.let {
+                payload[LOYALTY_STATUS_ERROR_SLUG] = it
+            }
+            errorMessage?.let {
+                payload[LOYALTY_STATUS_ERROR_MESSAGE] = errorMessage
+            }
+
+            return this
+        }
+
+        fun loyaltyPreAuthFailure(
+            slug: String?,
+            errorMessage: String?,
+            quoteId: String?,
+            correlationId: String?,
+            loyaltyMode: String,
+            balance: Int?
+        ): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+            correlationId?.let {
+                payload[CORRELATION_ID] = it
+            }
+            slug?.let {
+                payload[LOYALTY_STATUS_ERROR_SLUG] = it
+            }
+            balance?.let {
+                payload[LOYALTY_STATUS_BALANCE] = it
+            }
+            errorMessage?.let {
+                payload[LOYALTY_STATUS_ERROR_MESSAGE] = errorMessage
+            }
+
+            payload[LOYALTY_PREAUTH_TYPE] = loyaltyMode
+
+            return this
+        }
+
+        fun loyaltyPreAuthSuccess(
+            quoteId: String?,
+            correlationId: String?,
+            loyaltyMode: String?,
+            balance: Int?
```

Comment:
same question with `balance` as for `loyaltyPreAuthFailure()` @BartlomiejSopalaKarhoo

Reply:
Removed the balance, kept the loyaltyMode